shanoir-ng-studies/src/main/java/org/shanoir/ng/study/controler/StudyApi.java

```diff
@@ -223,4 +225,72 @@ void exportBIDSByStudyId(
+	@PreAuthorize("hasAnyRole('ADMIN', 'EXPERT', 'USER')")
+	ResponseEntity<List<DataUserAgreement>> getDataUserAgreementsByUserId(
+			@ApiParam(value = "id of the user", required = true) @PathVariable("userId") Long userId)
+			throws RestServiceException, IOException;
+	
+	@ApiOperation(value = "", notes = "Updates a data user agreement (DUA)", response = Void.class, tags = {})
+	@ApiResponses(value = {
+			@ApiResponse(code = 204, message = "dua updated", response = DataUserAgreement.class),
+			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
+			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
+			@ApiResponse(code = 422, message = "bad parameters", response = Void.class),
+			@ApiResponse(code = 500, message = "unexpected error", response = Void.class) })
+	@PutMapping(value = "/dua/{duaId}", produces = { "application/json" }, consumes = {"application/json" })
+	@PreAuthorize("hasAnyRole('ADMIN', 'EXPERT', 'USER')")
+	ResponseEntity<Void> acceptDataUserAgreement(
+			@ApiParam(value = "id of the dua", required = true) @PathVariable("duaId") Long duaId)
+			throws RestServiceException, MicroServiceCommunicationException;
+
+	@ApiOperation(value = "", notes = "Add DUA to a study", response = Void.class, tags = {})
+	@ApiResponses(value = {
+	        @ApiResponse(code = 200, message = "dua uploaded", response = Void.class),
+			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
+			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
+			@ApiResponse(code = 422, message = "bad parameters", response = ErrorModel.class),
+			@ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
+	@PostMapping(value = "dua-upload/{studyId}", produces = { "application/json" }, consumes = {
+			"multipart/form-data" })
+	@PreAuthorize("hasAnyRole('ADMIN', 'EXPERT') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_ADMINISTRATE')")
+	ResponseEntity<Void> uploadDataUserAgreement(
+			@ApiParam(value = "id of the study", required = true) @PathVariable("studyId") Long studyId,
+			@ApiParam(value = "file to upload", required = true) @Valid @RequestBody MultipartFile file)
+			throws RestServiceException;
+
+	@ApiOperation(value = "", notes = "Download DUA of a study", tags = {})
+	@ApiResponses(value = {
+	        @ApiResponse(code = 200, message = "dua downloaded", response = Resource.class),
+			@ApiResponse(code = 401, message = "unauthorized", response = Void.class),
+			@ApiResponse(code = 403, message = "forbidden", response = Void.class),
+			@ApiResponse(code = 422, message = "bad parameters", response = ErrorModel.class),
+			@ApiResponse(code = 500, message = "unexpected error", response = ErrorModel.class) })
+	@GetMapping(value = "dua-download/{studyId}/{fileName:.+}/")
+	@PreAuthorize("hasRole('ADMIN') or (hasAnyRole('EXPERT', 'USER') and @studySecurityService.hasRightOnStudy(#studyId, 'CAN_DOWNLOAD'))")
```

Comment:
If a user has the 'can see' right only, he does not have to sign the DUA ?

Reply:
good finding!!! I will change it!

Reply:
the dua download will require no rights, as I remove all rights during pending state "before acceptance"

Reply:
but the user should see it anyway

Reply:
changed